src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,205 @@
+                                "Format: description /by time");
+                    } else {
+                        throw new InvalidEventException("OOPS! The time is missing. " +
+                                "Format: description /by time");
+                    }
+                } else if (command.split("/by").length == 2) {
+                    if (command.split("/by")[0].equals("")) {
+                        throw new InvalidDeadlineException("OOPS! The description is missing. " +
+                                "Format: description /by time");
+                    }
+                }
+                String name = command.split(" /by")[0].trim();
+                String time = command.split(" /by")[1].trim();
+                if (isValidDate(time)) {
+                    LocalDate todoDate = LocalDate.parse(time);
+                    Deadline deadline = new Deadline(name, todoDate);
+                    output = taskList.addDeadline(deadline);
+                } else {
+                    Deadline deadline = new Deadline(name, time);
+                    output = taskList.addDeadline(deadline);
+                }
+            } else if (command.split(" ")[0].equals("event")) { // add event
+                command = command.substring(command.split(" ")[0].length() + 1, command.length()).trim();
+                if (command.split("/at").length == 0) {
+                    throw new InvalidEventException("OOPS! The description and time is missing. " +
+                            "Format: description /at time");
+                } else if (command.split("/at").length == 1) {
+                    if (command.split("/at")[0].equals("")) {
+                        throw new InvalidEventException("OOPS! The description and time is missing. " +
+                                "Format: description /at time");
+                    } else {
+                        throw new InvalidEventException("OOPS! The time is missing. " +
+                                "Format: description /at time");
+                    }
+                } else if (command.split("/at").length == 2) {
+                    if (command.split("/at")[0].equals("")) {
+                        throw new InvalidEventException("OOPS! The description is missing.");
+                    }
+                }
+                String name = command.split(" /at")[0].trim();
+                String time = command.split(" /at")[1].trim();
+                if (isValidDate(time)) {
+                    LocalDate todoDate = LocalDate.parse(time);
+                    Event event = new Event(name, todoDate);
+                    output = taskList.addEvent(event);
+                } else {
+                    Event event = new Event(name, time);
+                    output = taskList.addEvent(event);
+                }
+            } else {
+                throw new InvalidInputException();
+            }
+        return output;
+        }
```

Comment:
Hi, your parse function looks nice, and I think to make it more elegant, how about we use switch instead of if-else, as we can distinguish commands by their first word.